pkg/proc/core_linux_amd64_test.go

```diff
@@ -112,5 +120,26 @@ func TestSplicedReader(t *testing.T) {
 			}
 		})
 	}
+}
+
+func TestReadCore(t *testing.T) {
+	// This is all very fragile and won't work on hosts with non-default core patterns.
+	// Might be better to check in the core?
+	tempDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	fix := test.BuildFixture("panic")
+	bashCmd := fmt.Sprintf("cd %v && ulimit -c unlimited && GOTRACEBACK=crash %v", tempDir, fix.Path)
+	exec.Command("bash", "-c", bashCmd).Run()
+	corePath := path.Join(tempDir, "core")
 
+	core, err := readCore(corePath, fix.Path)
```

Comment:
How about checking if the core was produced first, and skipping the test if the core is not produced?
This test should test whether the core can be read, not whether the core is produced.